bark/world/evaluation/base_evaluator.hpp

```diff
@@ -19,7 +20,7 @@ class World;
 class ObservedWorld;
 namespace evaluation {
 
-typedef boost::variant<float, bool, std::string, int> EvaluationReturn;
+typedef boost::variant<float, bool, std::optional<bool>, std::string, int> EvaluationReturn;
```

Comment:
@juloberno would this extension be okay for you?
Why exactly do we use an optional here?

Reply:
Mhh, that seems counterintuitive to me. Why should it be optional to return something from an Evaluator?

Reply:
There are some situations that RSS cannot be checked, so I return None to Python if those situations happen. For example:

- The first second of the world (it is restricted by them, and don't know why..)
- The moment crossing roads and entering an intersection (for the last version of RSS)

Reply:
can't we just use an int? 0 for undefined, 1 for violated, 2 for not-violated?

Reply:
It is also possible, and up to be defined by us

Reply:
Okay, then i understand it. Is fine for me.